Installing-and-Configuring/OS/Generic.md

```diff
@@ -0,0 +1,118 @@
+
+If not using one of the included Operating Systems--where the vendor public key is [stored in the heads ROM](https://github.com/osresearch/heads/tree/master/initrd/etc/distro/keys)--you must verify the media immediately after download and sign it with your own key to establish a chain of trust.
+
+To verify an ISO with your key, create a partition with a Heads compatible filesystem on a USB storage device and copy the ISO image to it.  The most common filesystems are: FAT and ext. The layout might look like this:
+
+```shell
+/Qubes-R4.0-x86_64.iso
+/Qubes-R4.0-x86_64.iso.asc
+/Fedora-Workstation-Live-x86_64-27-1.6.iso
+/Fedora-Workstation-Live-x86_64-27-1.6.iso.sig
+/tails-amd64-3.7.iso
+/tails-amd64-3.7.iso.sig
+```
+
+Sign the ISO(s) with your own key:
+
+```shell
+gpg --output <iso_name>.sig --detach-sig <iso_name>
+```
+
+### Distro Specific ISO Boot
+
+Some distros require additional options for a successful boot directly from ISO.  See [Boot config files](#boot-config-files) for more information if your OS is not booting by selecting the default boot options from heads GUI.
+1. Boot from USB by either running `usb-scan` or reboot into USB boot mode (hit
+ 'u' before the normal boot)
+1. Select the install boot option for your distro of choice and work through the
+ standard OS installation procedures (including setting up LUKS disk encryption
+ if desired)
+1. Reboot and your new boot options should be available to be chosen by
+ selecting 'm' at the boot screen
+
+`kexec_iso_add.txt` and `kexec_iso_remove.txt` are useful to inject the
+ appropriate kernel arguments to allow it to load properly. ISOs for Debian
+ require that `kexec_iso_add.txt` contains the following to load properly:
+
+```text
+findiso=${ISO_PATH}
+```
+
+Take a look at [https://mbusb.aguslr.com/howto.html](https://mbusb.aguslr.com/howto.html)
+ for more variations on the distro-specific ISO mounting command lines
+ requirements.  By default Heads uses two variants of this when booting from
+ ISO where a `kexec_iso_add.txt` is not specified:
+
+```text
+fromiso=/dev/disk/by-uuid/$DEV_UUID/$ISO_PATH iso-scan/filename=/${ISO_PATH}
+```
+
+
+Installation Choices
+====
+
+See [OS](/OS/) for a review of various choices that will affect how you setup the Operating System.
+
+bootloader
+---
+
+heads reads /boot/grub.cfg in order to boot your system.  Since heads is acting as the bootloader you do not need to install grub to your MBR.  It won't hurt if you do since the MBR is mostly ignored.  If you choose to not install grub make sure you generate the grub.cfg file or heads cannot boot your system.
+
+Injecting LUKS key into OS boot
+----
+
+If using LUKS encryption on the root paritition you may need to modify the installation before booting will work to add the LUKS key into the process.
```

Comment:
"If using LUKS encryption on the root paritition you may need to modify the installation before booting will work to add the LUKS key into the process."

-> 

If using LUKS encryption on the root partition, you may need to modify your installation as prior of booting or adding TPM released LUKS key into the process.

Is that what you wanted to say @jtmoree-github-com ?